### PR TITLE
Add setting for minimum string length to trigger auto popup

### LINKED ISSUE #105: Allow setting minimum string length for auto popup
I'd like to be able to set an option for the minimum length of a typed string before auto popup; if set to `3`, for instance, then the auto popup won't appear until I've typed at least three characters.

This could be configurable by source, as well.

### PR DESCRIPTION
**Pull request by jsit**
Closes #105

### CODE REVIEW

**Comment by jsit** [comment]
I don’t think I understand? What do we need to check for?

**Comment by prabirshrestha** [comment]
https://github.com/prabirshrestha/asyncomplete.vim/blob/c7901012dd783429d1983a71a2dcc632bbbb03c2/autoload/asyncomplete.vim#L234

In v2 there are 3 ways to cause trigger.

1. You typed the trigger character such as `.`
2. Refresh pattern i.e. the word. 
3. Manually called force refresh

Currently your code is only handling 2 and not 1. I think handling 1 is important to make this useful.

**Comment by jsit** [comment]
Wouldn't any eligible source already get added to `l:sources_to_notify` from line `230`?:

https://github.com/prabirshrestha/asyncomplete.vim/blob/c7901012dd783429d1983a71a2dcc632bbbb03c2/autoload/asyncomplete.vim#L230-L239

**Comment by prabirshrestha** [comment]
Let us assume you are typing `server.restart_server()`. At which keystrokes do you want the autocomplete to popup?

Currently it popups at `ser` and `server.`, if `.` is added as a trigger character. Instead I think it should popup at `ser` and `server.res` i.e. after 3 chars have been typed after a trigger character or a word. If it is a force refresh it should always show.

**Comment by jsit** [comment]
Ah, that makes sense. I'll see if I can figure out how to make that work. Thanks!

File: autoload/asyncomplete.vim [hunk 1]
```diff
@@ -205,6 +205,16 @@ function! s:should_skip() abort
     endif
 endfunction
 
+function! s:get_min_chars(source_name) abort
+    let l:source = s:sources[a:source_name]
+    if has_key(l:source, 'min_chars')
```
Comment (prabirshrestha) [inline autoload/asyncomplete.vim]: We should be able to use `?:` instead

Comment (jsit) [reply autoload/asyncomplete.vim]: Fixed, thanks.

**Comment by prabirshrestha** [comment]
Seems like this got closed automatically because v2 branch got deleted and I can't reopen the PR. Could you send a new one directly to master.

**Pull request closed without merging**